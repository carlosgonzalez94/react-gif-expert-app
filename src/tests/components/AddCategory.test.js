import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe('Test of the component <AddCategory />', () => {
    const setCategories = jest.fn();
    let wrapper;

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });

    test('Should render the component AddCategory correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Should change the value of the input text', () => {
        const input = wrapper.find('input');    //obtengo el input
        const value = 'Hola Mundo';     //le envio el valor que voy a cambiar
        input.simulate('change', {      //le indico que simule el change
            target: {value}               //envio como parametro el target.value para que no se rompa el componente
        });
    });

    test('Should NOT send the info on submit', () => {
        wrapper.find('form').simulate('submit', {
            preventDefault(){}
        });

        expect(setCategories).not.toHaveBeenCalled();
    });

    test('Should call setCategories and clean the input ', () => {
        //simulate the value change of input
        const value = 'Hola Mundo';
        const input = wrapper.find('input');
        input.simulate('change', { target: {value} });

        //simulate the submit
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect(setCategories).toHaveBeenCalled();
        expect(input.prop('value')).toBe('');
    });
    
    
    
})
