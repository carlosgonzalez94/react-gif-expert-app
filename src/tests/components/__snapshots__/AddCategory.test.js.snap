// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of the component <AddCategory /> Should render the component AddCategory correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h1>
     
     
  </h1>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
